pkg/apply/applier.go

```diff
@@ -96,8 +98,21 @@ func (a *Applier) Run(ctx context.Context, invInfo inventory.Info, objects objec
 		}
 		validator.Validate(objects)
 
+		clusterInventory, err := a.invClient.Get(ctx, invInfo, inventory.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			clusterInventory = invInfo.NewEmptyInventory()
+		} else if err != nil {
+			handleError(eventChannel, err)
+			return
+		}
+		if clusterInventory.ID() != invInfo.ID() {
+			handleError(eventChannel, fmt.Errorf("expected inventory object to have inventory-id %q but got %q",
```

Comment:
Should we include the inventory object name/namespace in the error message?

What about the whole inventory-id annotation name? Or is that specific to the inventory type?

Will this error make sense when using a sharded client?